### ISSUE TITLE
Submerged assets can't be loaded (game stuck in loading screen)

### ISSUE THREAD

**Issue by MapleEve** [SubmergedAmongUs/Submerged#28]
**Describe the bug**
As known, the WW web is unstable in china, after I finally promote this map to local community, some ppl encounter the bug that the game is forever loading and canot login.

![4OU_`1`L~IX@_YE@39{8Y07](https://user-images.githubusercontent.com/1845053/170405353-f79b9154-cf29-49dd-ad16-324fcdbadc5d.jpg)

While EOS Manager cannot handle stable network connection and get puid or backend reply, it will stuck in this screen forever.

But, without submerged.dll, they can usually get into main screen and received a Offline notice and can still play as guest.

**To Reproduce**
host to mirror api.epicgames.dev to localhost, and try with submerged or without.

**Expected behavior**
They cannot stuck in loading screen forever.

**Screenshots**
above.

**Version (please complete the following information):**
 - Among Us: 3.29s
 - Submerged: 5.20
 - Crowded: own fork build with BE.559
 - TOR 4.1.4 : own fork build with BE.559

**Log file**
There's no error in Log file

**Additional context**
This issue is only happened in China with Submerged.dll, and under certain circumstances, but I can't find the new network structure code during de-comp, and since there's no source code for submerged, so I opened this issue.

**Comment by MapleEve** [comment]
[Info   : Submerged] Loading AssetLoader
[Info   : Submerged] Loading MapLoader
[Info   : Submerged] Loading TranslationController_Initialize_Patch
[Info   : Submerged] Loading StoreManager_Initialize_Patch
[Info   :   BepInEx] Loading [The Other Roles 4.1.4]
[Info   :Unhollower] Registered mono type TheOtherRoles.Modules.ModUpdateBehaviour in il2cpp domain
[Info   :Unhollower] Registered mono type BepInEx.IL2CPP.Utils.Collections.Il2CppManagedEnumerator in il2cpp domain
[Message:   BepInEx] Chainloader startup complete
[Fatal  : Submerged] System.NullReferenceException: Object reference not set to an instance of an object
  at Submerged.Loading.MonoBehaviours.AssetLoader+<Load>d__19.MoveNext () [0x00146] in <4e16d3ba5bc24bc7b144480d30f7dc01>:0 
[Info   : Submerged] Loaded TranslationController_Initialize_Patch
[Info   : Submerged] Loaded StoreManager_Initialize_Patch
[Info   : Submerged] Loaded MapLoader

**Comment by MapleEve** [comment]
[Info   : Submerged] Loading AssetLoader
[Info   : Submerged] Loading MapLoader
[Info   : Submerged] Loading TranslationController_Initialize_Patch
[Info   : Submerged] Loading StoreManager_Initialize_Patch
[Info   :   BepInEx] Loading [The Other Roles 4.1.4]
[Info   :Unhollower] Registered mono type TheOtherRoles.Modules.ModUpdateBehaviour in il2cpp domain
[Info   :Unhollower] Registered mono type BepInEx.IL2CPP.Utils.Collections.Il2CppManagedEnumerator in il2cpp domain
[Message:   BepInEx] Chainloader startup complete
[Fatal  : Submerged] System.NullReferenceException: Object reference not set to an instance of an object
  at Submerged.Loading.MonoBehaviours.AssetLoader+<Load>d__19.MoveNext () [0x00146] in <4e16d3ba5bc24bc7b144480d30f7dc01>:0 
[Info   : Submerged] Loaded TranslationController_Initialize_Patch
[Info   : Submerged] Loaded StoreManager_Initialize_Patch
[Info   : Submerged] Loaded MapLoader

**Comment by AlexejheroYTB** [comment]
Can you please attach the `TheOtherRoles.dll` and `Submerged.dll` files? (if they exist)

**Comment by AlexejheroYTB** [comment]
This issue doesn't seem like it's caused by internet connection. rather Submerged assets couldn't be loaded. This could be due to a corrupt Submerged installation, that's why I asked for the DLLs.

**Comment by MapleEve** [comment]
> This issue doesn't seem like it's caused by internet connection. rather Submerged assets couldn't be loaded. This could be due to a corrupt Submerged installation, that's why I asked for the DLLs.

TOR DLL: https://github.com/MapleEve/TheOtherRoles_RamboPlay/suites/6682220368/artifacts/253375332
TOR BE Frame: https://github.com/Eisbison/TheOtherRoles/releases/download/v4.1.4/TheOtherRoles.zip
Replaced with own translation fork.
Submerged: https://github.com/SubmergedAmongUs/Submerged/releases/download/v2022.5.20/Submerged.dll
independent dll

My local DLLs: https://drive.google.com/file/d/14NRce4MWSBe-wyF-ISKjaE4_4Y8PFfZz/view?usp=sharing

I think there's online fetch of assets for Submerged map but the CDN or OSS bracket is unavailable in China.

**Comment by AlexejheroYTB** [comment]
Submerged doesn't make any external network calls and works on among us official servers.  
As such, this likely isn't a network issue, and we are looking to see if it is related to the dlls you have currently.

Additionally, the logs you provided above trace this issue to loading the Submerged map assets from the local file.

**Comment by MapleEve** [comment]
> Submerged doesn't make any external network calls and works on among us official servers. As such, this likely isn't a network issue, and we are looking to see if it is related to the dlls you have currently.
> 
> Additionally, the logs you provided above trace this issue to loading the Submerged map assets from the local file.

Hmm, OK.
As long as removed Submerged.dll and the game can finally get into the login screen, than stuck in the loading screen.
And get dll back, it will stuck in the loading screen again.

So I suspect if there's some connection issue or so.

We had to make "TheOtherHats" pngs and json loaded locally than fetch through raw.github to avoid below issues we had encountered in China:
1. No mod logo in loading screen and BE doen't load.
2. Cannot join someone's room, auto DC
3. When a game finished, Host's version RPC failed and the others are not in the same version
